Add Docker

davisokoth opened the pull request:
This PR adds Docker for development purposes

davisokoth committed:
project: containerize app

Add Docker for development purposes

davisokoth committed:
project: eslint

Add `eslint` for typescript

crysfel commented:
This is great! No need to install azure in our local anymore, I will try to run it on my local before approving.

Could you fix the conflicts? Thanks!

brentmclark commented on Dockerfile:
```diff
@@ -1,6 +1,13 @@
+WORKDIR /home/node
+ENV AzureWebJobsScriptRoot=/home/node
+COPY package.json /home/node
+# RUN npm i -g copyfiles onchange rimraf typescript ttypescript
```
Is this here for future reference, or a leftover from local development?

crysfel replied:
Oh yes! let's update the readme. Now we only need two steps, right?

1. Download and install docker
2. docker-compose up

Well... that's what I'm thinking on running when testing this PR later today. @davisokoth let me know if there's anything else to do

davisokoth replied:
@Crysfel I should probably add a `docker-compose.yml` file so that its that simple. Will commit that tomorrow

davisokoth replied:
Will update `README.md` as well and fix the conflicts on `settings.json`

brentmclark reviewed (commented):
Aside from the `settings.json` conflict this LGTM.  Does anything need to change in the `README`?

davisokoth committed:
project: merge upstream development

Resolve `.vscode/settings.json` conflict

swissarmykirpan commented:
We should make sure it works with this: https://code.visualstudio.com/docs/remote/containers !

davisokoth committed:
project: add docker-compose file

davisokoth committed:
docker: fix file paths and names

davisokoth committed:
Merge branch 'development' into development

davisokoth commented:
> We should make sure it works with this: https://code.visualstudio.com/docs/remote/containers !

Alright. Let me see to that!

crysfel commented:
@davisokoth Awesome! I tested in my local and I didn't have any issues when running `docker-compose up`, it seems like it works fine.

The only thing I'm seeing is when calling any of the endpoints, I'm getting this error:

```
Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.add-user ---> Microsoft.Azure.WebJobs.Script.Rpc.RpcException: Result: Failure
coding-coach-api_1  | Exception: Worker was unable to load function add-user: 'Error: Cannot find module '@repositories/user-repository''
coding-coach-api_1  | Stack: Error: Cannot find module '@repositories/user-repository'
coding-coach-api_1  |     at Function.Module._resolveFilename (module.js:548:15)
coding-coach-api_1  |     at Function.Module._load (module.js:475:25)
coding-coach-api_1  |     at Module.require (module.js:597:17)
coding-coach-api_1  |     at require (internal/module.js:11:18)
coding-coach-api_1  |     at Object.<anonymous> (/home/node/app/dist/handlers/add-user/add-user-handler.js:29:27)
coding-coach-api_1  |     at Module._compile (module.js:653:30)
coding-coach-api_1  |     at Object.Module._extensions..js (module.js:664:10)
coding-coach-api_1  |     at Module.load (module.js:566:32)
coding-coach-api_1  |     at tryModuleLoad (module.js:506:12)
coding-coach-api_1  |     at Function.Module._load (module.js:498:3)
coding-coach-api_1  |    at Microsoft.Azure.WebJobs.Script.Description.WorkerLanguageInvoker.InvokeCore(Object[] parameters, FunctionInvocationContext context) in C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\Rpc\WorkerLanguageInvoker.cs:line 74
```

It seems like is not loading this module `@repositories/user-repository`, which TS should resolve.

davisokoth committed:
project: allow permissions to `.dotnet` folder

Fixes error in seeing `@repositories`

davisokoth committed:
project: prefer `tslint` over `eslint`

Add `tslint` as a dev-dependency
Enforce `tslint` requirements in various files

davisokoth committed:
Merge branch 'development' of https://github.com/davisokoth/coding-coach-api into development

Merge from origin

davisokoth committed:
Merge branch 'development' into development

davisokoth committed:
project: disable import ordering lint rule

Disable rule requiring ordering of imports to allow for `reflect-metadata` to be imported first

davisokoth committed:
Merge branch 'development' of https://github.com/davisokoth/coding-coach-api into development

davisokoth commented:
> @davisokoth Awesome! I tested in my local and I didn't have any issues when running `docker-compose up`, it seems like it works fine.
> 
> The only thing I'm seeing is when calling any of the endpoints, I'm getting this error:
> 
> ```
> Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.add-user ---> Microsoft.Azure.WebJobs.Script.Rpc.RpcException: Result: Failure
> coding-coach-api_1  | Exception: Worker was unable to load function add-user: 'Error: Cannot find module '@repositories/user-repository''
> coding-coach-api_1  | Stack: Error: Cannot find module '@repositories/user-repository'
> coding-coach-api_1  |     at Function.Module._resolveFilename (module.js:548:15)
> coding-coach-api_1  |     at Function.Module._load (module.js:475:25)
> coding-coach-api_1  |     at Module.require (module.js:597:17)
> coding-coach-api_1  |     at require (internal/module.js:11:18)
> coding-coach-api_1  |     at Object.<anonymous> (/home/node/app/dist/handlers/add-user/add-user-handler.js:29:27)
> coding-coach-api_1  |     at Module._compile (module.js:653:30)
> coding-coach-api_1  |     at Object.Module._extensions..js (module.js:664:10)
> coding-coach-api_1  |     at Module.load (module.js:566:32)
> coding-coach-api_1  |     at tryModuleLoad (module.js:506:12)
> coding-coach-api_1  |     at Function.Module._load (module.js:498:3)
> coding-coach-api_1  |    at Microsoft.Azure.WebJobs.Script.Description.WorkerLanguageInvoker.InvokeCore(Object[] parameters, FunctionInvocationContext context) in C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\Rpc\WorkerLanguageInvoker.cs:line 74
> ```
> 
> It seems like is not loading this module `@repositories/user-repository`, which TS should resolve.

Am still not sure how to fix this. Any pointers?

davisokoth committed:
project: add azurite `Dockerfile`

Add a`Dockerfile` to set up ans spin an `azurite` instance
